include/boost/algorithm/cxx11/copy_if.hpp

```diff
@@ -126,6 +126,82 @@ copy_until ( const Range &r, OutputIterator result, Predicate p )
     return boost::algorithm::copy_until (boost::begin (r), boost::end(r), result, p);
 }
 
+/// \fn copy_if_while ( InputIterator first, InputIterator last, OutputIterator result, CopyPredicate copy_pred, TerminatePred term_pred )
+/// \brief Copies all the elements from the input range that satisfy the
+/// predicate to the output range.
+/// \return The updated output iterator
+///
+/// \param first     The start of the input sequence
+/// \param last      One past the end of the input sequence
+/// \param result    An output iterator to write the results into
+/// \param copy_pred A predicate for testing whether to the current element
+/// \param term_pred A predicate for testing whether to end the copy operation
+template<typename InputIterator, typename OutputIterator, typename CopyPredicate, typename TerminatePred>
+BOOST_CXX14_CONSTEXPR std::pair<InputIterator, OutputIterator>
```

Comment:
I think the return type here and elsewhere should be `in_out_result`.  Users should expect to use `.in` and `.out` rather than `.first` and `.second`, as of C++20.  See https://en.cppreference.com/w/cpp/algorithm/ranges/return_types/in_out_result .  A version will have to be added to the `boost::algorithm` namespace.

Reply:
I disagree; I think that having `copy_while` return one type and having `copy_if_while` having a *different* return type (with the same information) will confuse users.

P.S I'm not really interested in changing the return type of `copy_while`, either.